NatLibFi/Skosmos
# Cyperess axe tests

## Reasons for creating this PR
To distinguish between web and cli accessibility tests

## Link to relevant issue(s), if any

- Closes #

## Description of the changes in this PR
Everything is new compared to the past. The intention is to run accessibility-related Axe tests under Cypress, while also offering developers the option to run the tests either in the browser or on the command line (command line implementation supports later GH Actions implementations).

## Known problems or uncertainties in this PR
The development of Skosmos and defining the usability tests are moving at different paces, so not all tests cases can be completed yet.

## How to
This is how you run cypress-axe tests:

In the root folder of Skosmos:

1. `npm install`

GUI Tests:

2. `test-accessibility-all-gui`
(Once the test browser has been opened and under the accessibility-tests folder)
- Open Inspector -> console -> clear the previous listing
- Click: axe-landing.cy.js (result table in Inspector)
... you can continue to run all the tests one by one in the same way .
- Close the browser and the CY application or run the command `ctrl-C` on the command line

CLI Tests:

3. (run tests on all page templates at the same time) `npm run test-accessibility-all-cli`
You can see the results in the table that appears in the command line listing
OR
Run accessibility tests separately per page template:
4. `npm run test-accessibility-about-cli`
`npm run test-accessibility-vocab-cli`
`npm run test-accessibility-feedback-cli`
`npm run test-accessibility-landing-cli`

## Checklist

- [ ] phpUnit tests pass locally with my changes
- [ ] I have added tests that show that the new code works, or tests are not relevant for this PR (e.g. only HTML/CSS changes)
- [x] The PR doesn't reduce accessibility of the front-end code (e.g. tab focus, scaling to different resolutions, use of `.sr-only` class, color contrast)
- [x] The PR doesn't introduce unintended code changes (e.g. empty lines or useless reindentation)
